src/Region/RegionHandler.cc

```diff
@@ -480,26 +571,36 @@ bool RegionHandler::SetStatsRequirements(
         return true;
     }
 
-    if (_regions.count(region_id)) {
-        // Save frame pointer
-        _frames[file_id] = frame;
+    if (!RegionSet(region_id)) {
+        spdlog::error("Statistics requirements failed: no region with id {}", region_id);
+        return false;
+    }
 
-        // Set requirements
-        ConfigId config_id(file_id, region_id);
-        _stats_req[config_id].stats_configs = stats_configs;
-        return true;
+    if (!IsClosedRegion(region_id)) {
+        spdlog::debug("Statistics requirements not valid for region {} type", region_id);
+        return false;
     }
-    return false;
+
+    // Save frame pointer
+    _frames[file_id] = frame;
+
+    // Set requirements
+    ConfigId config_id(file_id, region_id);
+    _stats_req[config_id].stats_configs = stats_configs;
+    return true;
 }
 
 void RegionHandler::RemoveRegionRequirementsCache(int region_id) {
     // Clear requirements and cache for a specific region or for all regions when closed
     if (region_id == ALL_REGIONS) {
         _histogram_req.clear();
         _stats_req.clear();
-        std::unique_lock<std::mutex> ulock(_spectral_mutex);
+        std::unique_lock<std::mutex> ulock1(_spectral_mutex);
         _spectral_req.clear();
-        ulock.unlock();
+        ulock1.unlock();
+        std::unique_lock<std::mutex> ulock2(_spatial_mutex);
+        _spatial_req.clear();
+        ulock2.unlock();
```

Comment:
This is very minor, but I would prefer to give these more descriptive names, like `ulock_spectral` and `ulock_spatial`.

Reply:
renamed to spectral_lock and spatial_lock